doc/guides/arch/arm.rst

```diff
@@ -0,0 +1,79 @@
+.. _arm_developer_guide:
+
+ARM Cortex-M Developer Guide
```

Comment:
Noting that this file is named `arm.rst`, the page title should probably be something like `ARM Developer Guide` with the Cortex-M-specific details in a subsection.

Reply:
I am not ready to work on the Cortex-R, though @stephanosio , so I wonder if i should just stick to the cortex-m user guide. I can rename the file to arm_cortex_m.rst, to make this more clear. Once we are ready to add a cortex-r guide, we can do file renaming

Reply:
Renamed this to arm_cortex_m.rst